src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java

```diff
@@ -447,6 +447,47 @@ public void succeedsWithValidTokenUsingDefaultClockAndHttpsDomain() {
                 .verify(token, "nonce");
     }
 
+    @Test
+    public void succeedsWhenOrganizationMatchesExpected() {
+        String token = JWT.create()
+            .withSubject("auth0|sdk458fks")
+            .withAudience(AUDIENCE)
+            .withIssuedAt(getYesterday())
+            .withExpiresAt(getTomorrow())
+            .withIssuer("https://" + DOMAIN + "/")
+            .withClaim("org_id", "org_123")
+            .sign(Algorithm.HMAC256("secret"));
+
+        String jwt = JWT.decode(token).getToken();
+
+        configureVerifier(jwt)
+            .withOrganization("org_123")
+            .build()
+            .verify(jwt);
+    }
+
+    @Test
+    public void failsWhenOrganizationDoesNotMatchExpected() {
+        exception.expect(IdTokenValidationException.class);
+        exception.expectMessage("Organization (org_id) claim mismatch in the ID token; expected \"org_abc\" but found \"org_123\"");
+
+        String token = JWT.create()
+            .withSubject("auth0|sdk458fks")
+            .withAudience(AUDIENCE)
+            .withIssuedAt(getYesterday())
+            .withExpiresAt(getTomorrow())
+            .withIssuer("https://" + DOMAIN + "/")
+            .withClaim("org_id", "org_123")
+            .sign(Algorithm.HMAC256("secret"));
+
+        String jwt = JWT.decode(token).getToken();
+
+        configureVerifier(jwt)
+            .withOrganization("org_abc")
+            .build()
+            .verify(jwt);
+    }
+
```

Comment:
Should we also add the following two tests:

- fails when an organization is provided to the IdTokenValidator but no claim was found
- Doesn't fail when no organization is provided to IdTokenValidator but the token has an org_id claim

Reply:
Yes, those are good tests to add, will push a new commit shortly.

Reply:
Additional tests to cover those scenarios pushed, good catch!